source/02-FN/03.ptx

```diff
@@ -11,6 +11,692 @@
+            <ol marker= "A." cols="2">
+              <li> <m>[4, -4]</m> </li>
+              <li> <m>[-4, 4]</m> </li>
+              <li> <m>[-5, 4]</m> </li>
+              <li> <m>[4, -5]</m> </li></ol></p>
+              <answer>
+                <p>
+                  Choice B: <m>[-4, 4]</m>
+                </p>
+              </answer>
+        </statement>
+      </task>
+    </activity>
+    
+    <activity xml:id="domain-and-range-of-ray">
+      <introduction>
+        <p>
+          Use the following graph to answer the questions below.
+        </p>
+        <figure xml:id="graph-of-ray">
+          <!-- <image source="Images/Graph-with-one-endpoint.jpg"/> -->
+          <image>
+            <sageplot>
+              p=arrow((0,4), (9,-2),xmin=-2,xmax=9,ymin=-5,ymax=6)
+              p+=point((0,4),pointsize=50,color='blue')
+              p
+            </sageplot>
+          </image>
+        </figure>
+      </introduction>
+      <task>
+        <statement>
+          <p>
+            What is the domain of this graph?
+            <ol marker= "A." cols="2">
+              <li> <m>[4, \infty)</m> </li>
+              <li> <m>(-\infty, 0]</m> </li>
+              <li> <m>(-\infty, 4]</m> </li>
+              <li> <m>[0, \infty)</m> </li></ol></p>
+              <answer>
+                <p>
+                  Choice D: <m>[0,\infty)</m>
+                </p>
+              </answer>
+        </statement>
+      </task>
+      <task>
+        <statement>
+          <p>
+            What is the range of this graph?
+            <ol marker= "A." cols="2">
+              <li> <m>[4, \infty)</m> </li>
+              <li> <m>(-\infty, 0]</m> </li>
+              <li> <m>(-\infty, 4]</m> </li>
+              <li> <m>[0, \infty)</m> </li></ol></p>
+              <answer>
+                <p>
+                  Choice C: <m>(-\infty, 4]</m>
+                </p>
+              </answer>
+        </statement>
+      </task>
+    </activity>
+
+    <remark>
+      When writing your intervals for domain and range, notice that you will need to write them from the smallest values to the highest values. For example, we wouldn't write <m>[4,-\infty)</m> as an interval because <m>-\infty</m> is smaller than <m>4</m>.
+      <p>
+        For domain, read the graph from left to right. For range, read the graph from bottom to top.
+      </p>
+    </remark>
+
+    <activity xml:id="domain-and-range-practice-ray-with-open-endpoint">
+      <introduction>
+        <p>
+          Use the following graph to answer the questions below.
+        </p>
+        <figure xml:id="graph-of-ray-with-open-endpoint">
+          <!-- <image source="Images/Ray-with-open-endpoint.jpg"/> -->
+            <sageplot>
+              p=arrow((3,-4), (-4,8),xmin=-6,xmax=5,ymin=-6,ymax=8)
+              p+=point((3,-4),pointsize=50,color='blue')
+              p+=point((3,-4),pointsize=10,color='white',zorder=2)
+              p
+            </sageplot>
+        </figure>
```

Comment:
Pretext throws a warning when `<sageplot>`s are not in an `<image>`.  Curious to see if the github action validates this.

Reply:
Looks like it gave a green check, but I saw the warning there when I went into the details.  I like that I can see the full PreTeXt output there

Reply:
yeah either the CLI should be erroring out there, or there should at least be a mode to run it that returns an error code. future work...